src/clients/translation/index.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import EnLocales from 'assets/translations/en.json';
+import i18next, { TFunctionKeys, ExistsFunction } from 'i18next';
+import {
+  initReactI18next,
+  useTranslation as useI18NextTranslation,
+  Trans as I18NextTrans,
+  TransProps as I18NextTransProps,
+} from 'react-i18next';
+
+i18next.use(initReactI18next).init({
+  resources: {
+    en: {
+      translation: EnLocales,
+    },
+  },
+  lng: 'en', // We only support English for now, but we'll need to detect the user's locale once we support more languages
+  fallbackLng: 'en',
+  interpolation: {
+    escapeValue: false,
+  },
+});
+
+interface TransProps extends Omit<I18NextTransProps<'t'>, 't' | 'i18nKey'> {
+  i18nKey: string;
+}
+
+export const useTranslation = () => {
+  const { t } = useI18NextTranslation();
+
+  const Trans: React.FC<TransProps> = ({ children, ...otherProps }) => (
+    <I18NextTrans t={t} {...otherProps}>
+      {children}
+    </I18NextTrans>
+  );
+
+  return {
+    t,
+    Trans,
+  };
+};
+
+// Only use this function when you need to render a string from outside a
+// component. Otherwise, use the t function or Trans component returned by the
+// useTranslation hook.
+export const t = (params: TFunctionKeys, values?: Record<string, unknown>) =>
+  i18next.t(params, values);
+
+// Function to check whether a translation exists
+export const translationExists: ExistsFunction = params => i18next.exists(params);
```

Comment:
I'm curious what the use case for this would be? When would we display something different then the default translation if it is missing?

Reply:
To be honest, your comment made me rethink about it and I think we should get rid of it.
I thought we could have cases when the translation key comes from a backend and so we'd need to check it exists before we call the `t` function, but that doesn't really apply actually since our contracts don't use error codes or anything we can use to identify errors in our translation files.